tests/test_zbhub.py

```diff
@@ -43,32 +43,34 @@ def test_receive_message(self):
         """
         # First, lets manually construct a Version message and send it into the Hub.
         message = {
-            'source_addr_long': b'\x00\ro\x00\x03\xbb\xb9\xf8',
-            'source_addr': b'\x88\x9f',
-            'source_endpoint': b'\x02',
-            'dest_endpoint': b'\x02',
             'profile': b'\xc2\x16',
-            'cluster': b'\x00\xf6',
-            'id': 'rx_explicit',
+            'source_addr': b'\x92T',
+            'dest_endpoint': b'\x02',
+            'rf_data': b'\t\x00\xfeT\x92\x1b\xf7r\x01\x00o\r\x009\x10\x07\x00\x01(\x00\x01\x0bAlertMe.com\tSmartPlug\n2011-07-25',
+            'source_endpoint': b'\x02',
             'options': b'\x01',
-            'rf_data': b'\tq\xfeMN\xf8\xb9\xbb\x03\x00o\r\x009\x10\x07\x00\x00)\x00\x01\x0bAlertMe.com\tSmartPlug\n2013-09-26'
+            'source_addr_long': b'\x00\ro\x00\x01r\xf7\x1b',
+            'cluster': b'\x00\xf6',
+            'id': 'rx_explicit'
         }
         self.hub_obj.receive_message(message)
         result = self.hub_obj.list_devices()
         expected = {
-            '00:0d:6f:00:03:bb:b9:f8': {
```

Comment:
Brilliant you have worked out the NodeId! 
From what you have seen do you think we could actually replace using the MAC address as the unique node ID key with the NodeId you have found?

Reply:
From what I see, the nodeid is actually the short addr, its just represented here in base10 not base16.  But there is more that need to be done on this 

- cross checks
- generate the status packet is still hardcoded for some things, and it should be fully encoding the mac / nodeid ......